src/MemoryBase.cs

```diff
@@ -63,189 +74,225 @@ public string ReadNullTerminatedString(int address)
-        /// <returns>Returns the long read from memory.</returns>
-        public float ReadSingle(int address)
+        /// <returns>Returns the single read from memory.</returns>
+        public float ReadSingle(StoreContext context, int address)
         {
             unsafe
             {
-                var i = ReadInt32(address);
+                var i = ReadInt32(context, address);
                 return *((float*)&i);
             }
         }
 
         /// <summary>
         /// Writes a single to memory.
         /// </summary>
+        /// <param name="context">The store context for the memory.</param>
         /// <param name="address">The zero-based address to write to.</param>
         /// <param name="value">The single to write.</param>
-        public void WriteSingle(int address, float value)
+        public void WriteSingle(StoreContext context, int address, float value)
         {
             unsafe
             {
-                WriteInt32(address, *(int*)&value);
+                WriteInt32(context, address, *(int*)&value);
             }
         }
 
         /// <summary>
         /// Reads a double from memory.
         /// </summary>
+        /// <param name="context">The store context for the memory.</param>
         /// <param name="address">The zero-based address to read from.</param>
         /// <returns>Returns the double read from memory.</returns>
-        public double ReadDouble(int address)
+        public double ReadDouble(StoreContext context, int address)
         {
             unsafe
             {
-                var i = ReadInt64(address);
+                var i = ReadInt64(context, address);
                 return *((double*)&i);
             }
         }
 
         /// <summary>
         /// Writes a double to memory.
         /// </summary>
+        /// <param name="context">The store context for the memory.</param>
         /// <param name="address">The zero-based address to write to.</param>
         /// <param name="value">The double to write.</param>
-        public void WriteDouble(int address, double value)
+        public void WriteDouble(StoreContext context, int address, double value)
         {
             unsafe
             {
-                WriteInt64(address, *(long*)&value);
+                WriteInt64(context, address, *(long*)&value);
             }
         }
 
         /// <summary>
         /// Grows the memory by the specified number of pages.
         /// </summary>
+        /// <param name="context">The store context for the memory.</param>
         /// <param name="delta">The number of WebAssembly pages to grow the memory by.</param>
-        /// <returns>Returns true if the memory grew or false if it would exceed the maximum size.</returns>
-        public bool Grow(uint delta)
+        /// <returns>Returns the previous size of the Webassembly memory, in pages.</returns>
+        public uint Grow(StoreContext context, uint delta)
```

Comment:
This may want to mention that this invalidates all previous spans?

Reply:
I'll update the comment for that.